pages/portal/[portalLink].js

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+
+import { Layout } from '../../components/Layout';
+
+const Portal = () => {
+  const router = useRouter();
+  const [nameRouter, setNameRouter] = useState();
+
+  useEffect(() => {
+    if (router.isReady) {
+      const { portalLink } = router.query;
+      setNameRouter(portalLink);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return <Layout>{nameRouter}</Layout>;
+};
```

Comment:
I wonder if the code could be simplified to this

```suggestion
const Portal = () => {
  const { query } = useRouter();
  
  return <Layout>{query.portalLink || ''}</Layout>;
};
```

It seems like an equivalent. Just without the setState rerender :D. 

Also, if you refresh the page. The `nameRouter` is never shown. Not sure what is going on

![CleanShot 2022-05-13 at 16 48 32](https://user-images.githubusercontent.com/12488648/168309221-dbe8c6fd-365d-400b-b854-b727e96ddcf7.png)

Reply:
No, it has to be in useEffect, because the router is not prepare at first load. I fixed the refresh.
```
const Portal = () => {
  const router = useRouter();

  const [nameRouter, setNameRouter] = useState();

  useEffect(() => {
    const { portalLink } = router.query;
    setNameRouter(portalLink);
  }, [router]);

  return (
    <Layout>
      <h1 className="headline--1 underline-large">{nameRouter}</h1>
    </Layout>
  );
};
export default Portal;
```